transaction/core/src/fog_hint.rs

```diff
@@ -117,6 +119,57 @@ impl FogHint {
         }
         FogHint::from_slice(&plaintext.as_ref()[0..RISTRETTO_PUBLIC_LEN])
     }
+
+    /// ct_decrypt
+    ///
+    /// Try to decrypt an encrypted payload onto this FogHint object in constant time.
+    /// Fails if decryption fails, or the magic number is wrong.
+    ///
+    /// # Arguments
+    /// * acct_server_private_key
+    /// * 128 byte encrypted payload
+    ///
+    /// # Returns
+    /// * Fog hint on success, cryptobox error otherwise
+    pub fn ct_decrypt(
+        ingest_server_private_key: &RistrettoPrivate,
+        ciphertext: &EncryptedFogHint,
+    ) -> (Self, bool) {
+        let mut default_plaintext = GenericArray::<
+            u8,
+            Diff<EncryptedFogHintSize, <VersionedCryptoBox as CryptoBox<Ristretto>>::FooterSize>,
+        >::default();
+
+        let mut rng = McRng::default();
+        let default_pubkey = RistrettoPublic::from_random(&mut rng);
+
+        default_plaintext.as_mut()[..RISTRETTO_PUBLIC_LEN]
+            .copy_from_slice(&default_pubkey.to_bytes());
+
+        let (plaintext, mut success) = match VersionedCryptoBox::default()
+            .decrypt_fixed_length(ingest_server_private_key, ciphertext.as_ref())
+        {
+            Ok(real_plaintext) => (real_plaintext, true),
+            Err(_) => (default_plaintext, false),
+        };
+
+        // Check magic numbers
+        for byte in &plaintext[RISTRETTO_PUBLIC_LEN..] {
+            if *byte != MAGIC_NUMBER {
+                success = false;
+            }
+        }
+
+        let default_fog_hint = FogHint::from_slice(
+            &default_plaintext.as_ref()[0..RISTRETTO_PUBLIC_LEN],
+        )
+        .expect("default plaintext should always contain a valid RistrettoPublic at the beginning");
```

Comment:
Should this return `(default_plaintext, false)`? Or it's ok to have the panic?

Reply:
I changed it to make the default fog hint directly from the default_pubkey, so it can't fail.